### ISSUE TITLE
Few question about testing with new images

### ISSUE THREAD

**Issue by mariiak2021** [nickgkan/orange#4]
Hi @nickgkan,

Can you please explain how can I get a grasping map for my own object image using the ORANGE algorithm? I'm new to it, so will appreciate the little explanation. :)

Thank you!

Best,
Mariia

**Comment by nickgkan** [comment]
Hi, I think a straightforward thing to do is to first train a model using depth images and store the weights. Then, you need the depth image of your object. You can load the model's weights and pass you depth image through it, it should output all grasping maps. The depth image should be converted to a tensor (check the loader on how to do this) with shape (1, 1, H, W).

For best results, try to use images that look similar to the Jacquard ones.

Best of luck!

**Comment by mariiak2021** [comment]
Hi @nickgkan 
Thank you for the explanation!
I have successfully created annotations for the jacquard dataset and trained the model using:
python main.py --dataset=jacquard --model=unet --use_bin_loss --use_bin_attention_loss --use_graspness_loss
At the training end I have received the file unet_jacquard.pt, which is saved in the 'models_path' directory = /models.

Now I would like to test the model with my depth image:
- I have converted the .tiff date image of my object to a tensor as you suggested
- I try to test the model unet_jacquard.pt, so calling:
CUDA_VISIBLE_DEVICES=-1 python main.py --dataset jacquard --net_name unet_jacquard  --im_size 224 
As I understood from the code the algorithm should check if the /models/unit_jacquard.pt exists. If it's exists, then testing should start automatically?

But in my case it's treated like not existed file, so instead the algorithm starting to training:
2021-09-22 17:32:32,164 - INFO - Performing training for unet_jacquard_jacquard

Can you please help me?
Also I don't understand how to pass my depth image inside for testing... Will appreciate your advice!

**Comment by nickgkan** [comment]
Hi @mariiak2021
Good to hear that you trained the model successfully.

On the training log you're observing: this is expected, but soon after you should see another log "Found existing trained model." It will then evaluate on the test set of Jacquard.

Now it should be easy to write a script to test your model on any depth image you want. Some code to use (please fill the missing imports etc):

```
# Load model and checkpoint
net = UNet(1, 3)
model_path_name = 'models/unet_best2_grasp_point_jacquard.pt'
chkpt = torch.load(model_path_name)
net.load_state_dict(chkpt['model_state_dict'], strict=False)
net = net.cpu()

# Load your depth image, i.e. implement the following function
depth_img = get_depth_image()
depth_img = torch.clamp(depth_img - depth_img.mean(), -1, 1)  # please apply the same preprocessing you used for training

# Forward pass and plots
with torch.no_grad():
    pos, cos, sin, width, graspness, bins = net(depth_img.unsqueeze(0))
pos_g = pos * torch.sigmoid(graspness.unsqueeze(1))
pos_f = pos_g * torch.sigmoid(bins)
print('Pos')
for i, p in enumerate(pos.squeeze()):
    my_plot(p, img, alpha=0.5, name='pos' + str(i) + '.png')
print('Pos with graspness')
for i, p in enumerate(pos_g.squeeze()):
    my_plot(p, img, alpha=0.5, name='pos_w_graspness' + str(i) + '.png')
print('Pos with graspness and bins')
for i, p in enumerate(pos_f.squeeze()):
    my_plot(p, img, alpha=0.5, name='pos_w_graspness_n_bins' + str(i) + '.png')
```

The above should be the base of your script, you'll need to import torch, matplotlib etc, as well set some variables. Try to ensure that the same preprocessing is applied, i.e. same size, normalization with mean, clamping. See the dataset class for that as well.
Again, best of luck!
